Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/util/CoreHelpUtils.java

```diff
@@ -118,4 +131,110 @@ public static Date getDate(String dateString) {
+                address.getState(), address.getPostal()));
+        }
+        if (isId(patient.getPatientId())) {
+            doc.setPatientRecordId(patient.getPatientId());
+        }
+    }
+
+    public static void updateDocumentBy(DocumentMetadataType doc, PatientType patientType) {
+        doc.setPatientId(getPatientIdentifierIso(patientType));
+        doc.setSourcePatientId(getPatientIdentifierIso(patientType));
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd");
+
+        // setPIDs
+        PersonNameType personname = lastItem(patientType.getPersonNameList());
+        if (personname != null) {
+            doc.setPid5(MessageFormat.format("{0}^{1}^^^", personname.getLastName(), personname.getFirstName()));
+        }
+        Date dateOfBirth = getDate(patientType.getDateOfBirth());
+        if (dateOfBirth != null) {
+            doc.setPid7(sdf.format(dateOfBirth));
+        }
+        doc.setPid8(patientType.getGender());
+
+        AddressType address = lastItem(patientType.getAddressList());
+        if (address != null) {
+            doc.setPid11(MessageFormat.format("{0}^^{1}^{2}^{3}^", address.getStreet1(), address.getCity(),
+                address.getState(), address.getPostal()));
+        }
+        if (isId(patientType.getPatientId())) {
+            doc.setPatientRecordId(patientType.getPatientId());
+        }
+    }
+
+    public static <T> T lastItem(List<T> items) {
+        T item = null;
+        if (CollectionUtils.isNotEmpty(items)) {
+            item = items.get(items.size() - 1);
+        }
+        return item;
+    }
+
+    public static <T> T firstItem(List<T> items) {
+        T item = null;
+        if (CollectionUtils.isNotEmpty(items)) {
+            item = items.get(0);
+        }
+        return item;
+    }
+
+    public static boolean isId(Long id) {
+        return id != null && id.longValue() > 0L;
+    }
+
+    public static String getPatientIdentifier(PatientType patient) {
+        PersonNameType name = lastItem(patient.getPersonNameList());
+        IdentifierType identifier = lastItem(patient.getIdentifierList());
+        if (null != name && null != identifier) {
+            return MessageFormat.format("{0} {1} - {2}", name.getFirstName(), name.getLastName(), identifier.getId());
+        }
+        return null;
+    }
+
+    public static String getPatientIdentifierIso(PatientType patient) {
+        IdentifierType identifier = lastItem(patient.getIdentifierList());
+        if (null != identifier) {
+            return MessageFormat.format("{0}^^^&{1}&ISO", identifier.getId(), identifier.getOrganizationId());
+        }
+        return null;
+    }
+
+    public static <R, T> List<R> castListType(List<T> patientList) {
+        List<R> list = new ArrayList<>();
+        for (T patient : patientList) {
+            list.add((R) patient);
+        }
+        return list;
+    }
+
+    public static <K, V> boolean isCollectionEmpty(Map<K, V> map) {
```

Comment:
Why we are recreating CollectionUtils.isEmpty()?

Reply:
going to use maputils